creme/creme_core/core/enumerable.py

```diff
@@ -94,8 +95,33 @@ def _queryset(self):
 
         return qs.complex_filter(limit_choices_to) if limit_choices_to else qs
 
-    def choices(self, user):
-        return [*map(self.instance_as_dict, self._queryset())]
+    def filter_by_term(self, user, term, limit=None):
+        queryset = self._queryset()
+        term = term.lower()
+
+        choices = filter(
+            lambda c: term in c['label'].lower(),
+            map(self.instance_as_dict, queryset)
+        )
+
+        return list(islice(choices, limit) if limit else choices)
+
+    def filter_by_values(self, user, values, limit=None):
+        queryset = self._queryset().filter(id__in=values)
+        return list(
+            map(self.instance_as_dict, queryset[:limit] if limit else queryset)
+        )
```

Comment:
It seems this method could return a QuerySet, and `choices()` does the `map()` & `limit` work